pallets/relayer/src/lib.rs

```diff
@@ -0,0 +1,303 @@
+// This file is part of the Polymesh distribution (https://github.com/PolymathNetwork/Polymesh).
+// Copyright (c) 2020 Polymath
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3.
+
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+//! # Simple Relayer Module
+//!
+//! TODO: Add pallet description.
+//! TODO: Add tests.
+//! TODO: Add support for `AuthorizationData::AddRelayerPayingKey` to `CddAndFeeDetails` in `pallets/runtime/*/src/fee_details.rs`
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(feature = "runtime-benchmarks")]
+pub mod benchmarking;
+
+use codec::{Decode, Encode};
+use frame_support::{decl_error, decl_module, decl_storage, dispatch::DispatchResult, ensure};
+use pallet_identity::{self as identity, PermissionedCallOriginData};
+pub use polymesh_common_utilities::traits::relayer::{
+    Config, Event, IdentityToRelayer, RawEvent, WeightInfo,
+};
+use polymesh_primitives::{AuthorizationData, AuthorizationError, IdentityId, Signatory};
+
+type Identity<T> = identity::Module<T>;
+
+/// The paying key and remaining polyx balance.
+#[derive(Encode, Decode, Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Subsidy<Acc, Bal> {
+    pub paying_key: Acc,
+    pub remaining: Bal,
+}
+
+decl_storage! {
+    trait Store for Module<T: Config> as Relayer {
+        /// map `user_key` to `paying_key`
+        pub Subsidies get(fn paying_keys):
+            map hasher(blake2_128_concat) T::AccountId => Subsidy<T::AccountId, T::Balance>;
```

Comment:
```suggestion
            map hasher(blake2_128_concat) T::AccountId => Option<Subsidy<T::AccountId, T::Balance>>;
```
(Handles absence a bit cleaner in the type system.)